core/trino-main/src/main/java/io/trino/server/security/jwt/JwtAuthenticator.java

```diff
@@ -51,15 +53,22 @@ public JwtAuthenticator(JwtAuthenticatorConfig config, SigningKeyResolver signin
             jwtParser.requireAudience(config.getRequiredAudience());
         }
         this.jwtParser = jwtParser.build();
+        userMapping = createUserMapping(config.getUserMappingPattern(), config.getUserMappingFile());
     }
 
     @Override
-    protected Optional<Principal> extractPrincipalFromToken(String token)
+    protected Optional<Identity> createIdentity(String token)
+            throws UserMappingException
     {
-        return Optional.ofNullable(jwtParser.parseClaimsJws(token)
+        Optional<String> principal = Optional.ofNullable(jwtParser.parseClaimsJws(token)
                 .getBody()
-                .get(principalField, String.class))
-                .map(BasicPrincipal::new);
+                .get(principalField, String.class));
+        if (principal.isEmpty()) {
+            return Optional.empty();
+        }
+        return Optional.of(Identity.forUser(userMapping.mapUser(principal.get()))
+                .withPrincipal(new BasicPrincipal(principal.get()))
+                .build());
```

Comment:
You can continue the chain and avoid the if statement here by using `map`.

Reply:
I'm afraid it would not work. User mapping throws `UserMappingException` which  is checked exception.